lib/active_shipping/carriers/ups.rb

```diff
@@ -816,16 +816,18 @@ def parse_tracking_response(response, options = {})
         # Build status hash
         status_nodes = first_package.css('Activity > Status > StatusType')
 
-        # Prefer a delivery node
-        status_node = status_nodes.detect { |x| x.at('Code').text == 'D' }
-        status_node ||= status_nodes.first
+        if status_nodes.present?
```

Comment:
If we're already using `nil` as a status, then it's 👍 

We really should avoid `nil` in another PR though, and have an `:unknown` status so there's no room for interpretation of a `nil` status.